test/soak/soak_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+package soak
+
+import (
+	"fmt"
+	"log"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSoakLow(t *testing.T) {
+	switch runtime.GOOS {
+	case "darwin":
+		// todo:
+	case "linux":
+		runTest(t, "SoakTestLowLinux", "resources/soak_linux.json", 5, 100_000_000)
+	case "windows":
+		runTest(t, "SoakTestLowLinux", "resources/soak_windows.json", 5, 100_000_000)
+	}
+}
+
+// todo: add high througput
+// todo: logrotate
+// todo: multiple-logs
+func TestSoakHigh(t *testing.T) {
+	switch runtime.GOOS {
+	case "darwin":
+		// todo:
+	case "linux":
+		runTest(t, "SoakTestHighLinux", "resources/soak_linux.json", 50, 200_000_000)
+	case "windows":
+		runTest(t, "SoakTestHighWindows", "resources/soak_windows.json", 50, 200_000_000)
```

Comment:
cpu limit = 50 and mem limit = 200M will likely change.
I have not run against windows with this specific workload yet.